include/random_walks/new_gaussian_accelerated_billiard_walk.hpp

```diff
@@ -0,0 +1,274 @@
+                coef = 1.0;
+
+                vEv = (_v.getCoefficients().transpose() * E.template selfadjointView<Eigen::Upper>()).dot(_v.getCoefficients());
+
+                Point p0 = _p;
+
+                it = 0;
+                std::pair<NT, int> pbpair;
+                if(!was_reset) {
+                    pbpair = P.line_positive_intersect(_p, _v, _lambdas, _Av, _lambda_prev, _update_parameters);
+                } else {
+                    pbpair = P.line_first_positive_intersect(_p, _v, _lambdas, _Av, _update_parameters);
+                    was_reset = false;
+                }                                                    
+
+                if (T <= pbpair.first) {
+                    _p += (T * _v);
+                    _lambda_prev = T;
+                    continue;
+                }
+
+                _lambda_prev = dl * pbpair.first;
+                _p += (_lambda_prev * _v);
+                T -= _lambda_prev;
+                coef = P.compute_reflection(_v, _p, _AE, _AEA, vEv, _update_parameters);
+                it++;
+
+
+                while (it < _rho)
+                {
+                    std::pair<NT, int> pbpair
+                        = P.line_positive_intersect(_p, _v, _lambdas, _Av, _lambda_prev, _AA, _update_parameters);
+                    _Av *= coef;
+                    _update_parameters.inner_vi_ak *= coef;
+                    pbpair.first /= coef;
+
+                    if (T <= pbpair.first) {
+                        _p += (T * _v);
+                        _lambda_prev = T;
+                        break;
+                    }
+                    _lambda_prev = dl * pbpair.first;
+                    _p += (_lambda_prev * _v);
+                    T -= _lambda_prev;
+                    coef = P.compute_reflection(_v, _p, _AE, _AEA, vEv, _update_parameters);
+                    it++;
+                }
+                if (it == _rho){
+                    _p = p0;
+                    was_reset = true;
+                }
+            }
+
+            p = _p;
+        }
+
+    private :
+
+        template <typename GenericPolytope>
+        inline void initialize(GenericPolytope& P,
+                               Point const& p,  // a point to start
+                               MT const& E,   // covariance matrix representing the Gaussian distribution
+                               RandomNumberGenerator &rng)
+        {
+            unsigned int n = P.dimension();
+            const NT dl = 0.995;
+            was_reset = false;
+            _lambdas.setZero(P.num_of_hyperplanes());
+            _Av.setZero(P.num_of_hyperplanes());
+            _p = p;
+            _AE.noalias() = P.get_mat() * E;
+            _AEA.resize(P.num_of_hyperplanes());
+            for(int i = 0; i < P.num_of_hyperplanes(); ++i)
+            {
+                _AEA(i) = _AE.row(i).dot(P.get_mat().row(i));
+            }
```

Comment:
This probably is equivalent with
_AEA = _AE.cwiseProduct(P.get_mat()).rowwise().sum();

Plz check.